test/unit/org/apache/cassandra/cql3/CQLTester.java

```diff
@@ -1031,122 +1034,222 @@ protected void alterTableMayThrow(String query) throws Throwable
+    public void waitForTableIndexesQueryable(String keyspace, String table)
     {
-        long start = currentTimeMillis();
-        boolean indexCreated = false;
-        while (!indexCreated)
-        {
-            Object[][] results = getRows(execute("select index_name from system.\"IndexInfo\" where table_name = ?", keyspace));
-            for(int i = 0; i < results.length; i++)
-            {
-                if (index.equals(results[i][0]))
-                {
-                    indexCreated = true;
-                    break;
-                }
-            }
+        waitForAssert(() -> assertTrue(areTableIndexesQueryable(keyspace, table)), 60, TimeUnit.SECONDS);
+    }
 
-            if (currentTimeMillis() - start > 5000)
-                break;
+    public void waitForIndexQueryable(String index)
+    {
+        waitForIndexQueryable(KEYSPACE, index);
+    }
 
-            Thread.sleep(10);
-        }
+    /**
+     * Index creation is asynchronous, this method waits until the specified index is queryable.
+     *
+     * @param keyspace the index keyspace name
+     * @param index the index name
+     */
+    public void waitForIndexQueryable(String keyspace, String index)
+    {
+        waitForAssert(() -> assertTrue(isIndexQueryable(keyspace, index)), 60, TimeUnit.SECONDS);
+    }
 
-        return indexCreated;
+    protected void waitForIndexBuilds(String index)
+    {
+        waitForIndexBuilds(KEYSPACE, index);
     }
 
     /**
      * Index creation is asynchronous, this method waits until the specified index hasn't any building task running.
      * <p>
-     * This method differs from {@link #waitForIndex(String, String, String)} in that it doesn't require the index to be
-     * fully nor successfully built, so it can be used to wait for failing index builds.
+     * This method differs from {@link #waitForIndexQueryable(String, String)} in that it doesn't require the
+     * index to be fully nor successfully built, so it can be used to wait for failing index builds.
      *
      * @param keyspace the index keyspace name
-     * @param indexName the index name
-     * @return {@code true} if the index build tasks have finished in 5 seconds, {@code false} otherwise
+     * @param index the index name
      */
-    protected boolean waitForIndexBuilds(String keyspace, String indexName) throws InterruptedException
+    protected void waitForIndexBuilds(String keyspace, String index)
+    {
+        waitForAssert(() -> assertFalse(isIndexBuilding(keyspace, index)), 60, TimeUnit.SECONDS);
+    }
+
+    protected boolean areTableIndexesQueryable()
     {
-        long start = currentTimeMillis();
-        SecondaryIndexManager indexManager = getCurrentColumnFamilyStore(keyspace).indexManager;
+        return areTableIndexesQueryable(KEYSPACE, currentTable());
+    }
 
-        while (true)
+    protected boolean areTableIndexesQueryable(String keyspace, String table)
```

Comment:
The test failures we're seeing lead me to believe that we should actually have this method return the set of indexes that haven't yet become queryable. That might give us some insight into what's going on if the assertion times out. (In other words, we can use `assertEquals(<empty set>, areTableIndexesQueryable(...))` in `waitForTableIndexesQueryable()` and get more information on failure to help debug.)

Reply:
https://app.circleci.com/pipelines/github/adelapena/cassandra/2892/workflows/837b186e-6cd9-442c-baaf-8bc067f56b21/jobs/43816/tests#failed-test-0

Reply:
I have replaced it by a `getNotQueryableIndexes` method.

Reply:
It seems the failure can be reproduced on the base feature branch too: https://app.circleci.com/pipelines/github/adelapena/cassandra/2894/workflows/71d2d463-2780-496d-a6fd-8c530768ae1b/jobs/44121/tests

Reply:
Yeah, I'm fine merging w/ `getNotQueryableIndexes()` in place and then we'll at least get a more helpful error when this fails again.